src/variety/squarejam.js

```diff
@@ -0,0 +1,168 @@
+//
+//  squarejam.js
+//
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["squarejam"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		inputModes: { edit: ["number", "clear"], play: ["border", "subline"] },
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					if (this.btn === "left" && this.isBorderMode()) {
+						this.inputborder();
+					} else {
+						this.inputQsubLine();
+					}
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart) {
+					this.inputqnum();
+				}
+			}
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true
+	},
+
+	Cell: {
+		minnum: 1,
+		maxnum: 999,
```

Comment:
Maxnum should be a number between 1 and the smallest grid dimension.

Reply:
I wasn't sure what happens if you modify the size dynamically, so I just left it at this since a few other puzzles have this behavior (although not ones that are unique to pzprxs). I'll do some testing and see what happens

Reply:
`maxnum` can be either a constant number, or a function that returns a number. It's only used as a suggestion when entering numbers into a cell, it doesn't actually guarantee that numbers stay within range. (e.g. you can type a number, then shrink the grid down. The number will stay the same, even though it can no longer be satisfied)